physics/sfc_drv_ruc.F90

```diff
@@ -16,29 +16,208 @@ module lsm_ruc
       character(len=*),     intent(out) :: errmsg
       integer,              intent(out) :: errflg
 
+! --- local
+      integer  :: ipr, i, k
+      logical  :: debug_print
+      logical, dimension(im) :: land, icy, wet
+      integer, dimension(im) :: soiltyp, vegtype
+      real (kind=kind_phys), dimension(im) :: frland
+
       ! Initialize CCPP error handling variables
       errmsg = ''
       errflg = 0 
 
+      ipr = 10
+      debug_print = .false.
+
+!> - Call rucinit() to initialize soil/ice/water  variables
+
+      if ( debug_print) then
+        write (0,*) 'RUC LSM initialization'
+        write (0,*) 'lsoil_ruc, lsoil',lsoil_ruc, lsoil
+        write (0,*) 'me, isot, ivegsrc, nlunit ',me, isot, ivegsrc, nlunit
+        write (0,*) 'noah soil temp',stc(ipr,:)
+        write (0,*) 'noah mois(ipr)',ipr,smc(ipr,:)
+        write (0,*) 'stype=',stype(ipr)
+        write (0,*) 'vtype=',vtype(ipr)
+        write (0,*) 'fice=',fice(ipr)
+        write (0,*) 'tsfc_lnd=',tsfc_lnd(ipr)
+        write (0,*) 'tsfc_wat=',tsfc_wat(ipr)
+        write (0,*) 'tg3=',tg3(ipr)
+        write (0,*) 'slmsk=',slmsk(ipr)
+        write (0,*) 'flag_init =',flag_init
+        write (0,*) 'flag_restart =',flag_restart
+      endif
+
       !--- initialize soil vegetation
       call set_soilveg_ruc(me, isot, ivegsrc, nlunit)
 
+      soiltyp(:) = 0
+      vegtype(:) = 0
+      land (:) = .false.
+      icy (:)  = .false.
+      wet (:)  = .false.
+
+      if (frac_grid) then  ! fice is ice fraction wrt water area
+        do i=1,im
+          frland(i) = landfrac(i)
+          if (frland(i) > zero) land(i) = .true.
+          if (frland(i) < one) then
+            if (flag_cice(i)) then
+              if (fice(i) >= min_seaice) then
+                icy(i)  = .true.
+              else
+                fice(i) = zero
+              endif
+            else
+              if (fice(i) >= min_lakeice) then
+                icy(i) = .true.
+              else
+                fice(i) = zero
+              endif
+            endif
+            if (fice(i) < one ) then
+              wet(i)=.true. ! some open ocean/lake water exists
+            end if
+          else
+            fice(i) = zero
+          endif
+        enddo
+
+      else
+
+        do i = 1, IM
+          frland(i) = zero
+          if (slmsk(i) == 0) then
+            wet(i)  = .true.
+            fice(i) = zero
+          elseif (slmsk(i) == 1) then
+            land(i)   = .true.
+            frland(i) = one
+            fice(i)   = zero
+          else
+            icy(i)  = .true.
+            if (fice(i) < one) then
+              wet(i) = .true.
+            endif
+          endif
+        enddo
+
+      endif
+
+      do i  = 1, im ! i - horizontal loop
+        if( land(i) ) then
+        !-- land
+          soiltyp(i)  = int( stype(i)+0.5 )
+          vegtype(i)  = int( vtype(i)+0.5 )
+        elseif( icy(i) > 0. ) then
```

Comment:
this line must be
```
elseif( icy(i)) then
```
because icy is a logical, not an integer

Reply:
Dom, I have corrected the error. 
Somehow this error did not show in the compilation.
Thank you!

Reply:
Only the GNU compiler complained about it, Intel ignored it. Thanks for fixing it so quickly.